### PR TITLE
Close runCommand stdout using communicate

### PR DESCRIPTION
**Pull request by scop**
https://twistedmatrix.com/trac/ticket/8640

Split to separate ticket/PR as instructed in #363

### CODE REVIEW

**Comment by rodrigc** [comment]
This looks good, but can you add a topfile?

File: twisted/python/_release.py [hunk 1]
```diff
@@ -58,8 +58,8 @@ def runCommand(args, cwd=None):
     @raise CommandFailed: when the program exited with a non-0 exit code.
     """
     process = Popen(args, stdout=PIPE, stderr=STDOUT, cwd=cwd)
-    stdout = process.stdout.read()
-    exitCode = process.wait()
+    stdout = process.communicate()[0]
+    exitCode = process.poll()
```
Comment (rodrigc) [inline twisted/python/_release.py:62]: Why did you change process.wait() to process.poll() ?

Comment (markrwilliams) [inline twisted/python/_release.py:62]: Popen.communicate (on the line before) until the child process exits.
Popen.poll lets you retrieve the exit status without blocking.

https://docs.python.org/2/library/subprocess.html#subprocess.Popen.communicate

https://docs.python.org/2/library/subprocess.html#subprocess.Popen.poll

Comment (rodrigc) [inline twisted/python/_release.py:62]: Ah OK, thanks for pointing that out Popen.communicate() waits for the child process to exit,
that's good.
